ocsp/src/ocsp.rs

```diff
@@ -0,0 +1,463 @@
+use der::asn1::{BitString, Ia5String, ObjectIdentifier, OctetString, UIntBytes};
+use der::asn1::{GeneralizedTime, Null};
+use der::{Any, Choice, Enumerated, Sequence};
+use spki::AlgorithmIdentifier;
+use x509::ext::pkix::name::GeneralName;
+use x509::ext::pkix::{AuthorityInfoAccessSyntax, CrlReason};
+use x509::ext::Extensions;
+use x509::name::Name;
+use x509::Certificate;
+
+/// ```text
+/// OCSPRequest ::= SEQUENCE {
+///    tbsRequest              TBSRequest,
+///    optionalSignature   [0] EXPLICIT Signature OPTIONAL }
+/// ```
+/// [RFC 6960 Section 4.1.1]: https://datatracker.ietf.org/doc/html/rfc6960#section-4.1.1
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+pub struct OcspRequest<'a> {
+    pub tbs_request: TbsRequest<'a>,
+
+    #[asn1(context_specific = "0", optional = "true", tag_mode = "EXPLICIT")]
+    pub optional_signature: Option<Signature<'a>>,
+}
+
+/// ```text
+/// TBSRequest ::= SEQUENCE {
+///    version             [0] EXPLICIT Version DEFAULT v1,
+///    requestorName       [1] EXPLICIT GeneralName OPTIONAL,
+///    requestList             SEQUENCE OF Request,
+///    requestExtensions   [2] EXPLICIT Extensions OPTIONAL }
+/// ```
+/// [RFC 6960 Section 4.1.1]: https://datatracker.ietf.org/doc/html/rfc6960#section-4.1.1
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+pub struct TbsRequest<'a> {
+    ///    version             \[0\] EXPLICIT Version DEFAULT v1,
```

Comment:
Re: chunking things up, I asked that very question yesterday before sending the PR and it was agreed not to do that. The reason for sending in one chunk is certval tests cover a lot of territory but need the full feature range. 

Re: documentation, we could modify to align with your tastes now or later. I'd like to get this backlog that's been growing since December cleared then address minor nits personally. There will be a few suggestions noted in chat to make as well to get at parsing errors.

Reply:
Yes, it was a private chat. Re: certval, this is the library I have mentioned many times and provided a link to. The currently posted one is well out of date (late January). I aim to update it once this clears. 

I can go through and delete the doc comments to match your preferred style. I would prefer to leave the CMS comments (i.e., unimpl'ed structs) but don't want to hold anything up over it. 

Will be out for a bit. Expect delays in further replies.